client/js/src/client.ts

```diff
@@ -341,24 +341,39 @@ export function api_factory(fetch_implementation: typeof fetch): Client {
 			): Promise<unknown> {
 				let data_returned = false;
 				let status_complete = false;
+				let dependency;
+				if (typeof endpoint === "number") {
+					dependency = config.dependencies[endpoint];
+				} else {
+					const trimmed_endpoint = endpoint.replace(/^\//, "");
+					dependency = config.dependencies[api_map[trimmed_endpoint]];
+				}
+
+				if (dependency.types.continuous) {
+					throw new Error(
+						"Cannot call predict on this function as it may run forever. Use submit instead"
+					);
+				}
+
 				return new Promise((res, rej) => {
 					const app = submit(endpoint, data, event_data);
+					let result;
 
 					app
 						.on("data", (d) => {
-							data_returned = true;
 							if (status_complete) {
 								app.destroy();
+								data_returned = true;
+								res(d);
 							}
-							res(d);
+							result = d;
 						})
 						.on("status", (status) => {
 							if (status.stage === "error") rej(status);
-							if (status.stage === "complete" && data_returned) {
-								app.destroy();
-							}
 							if (status.stage === "complete") {
 								status_complete = true;
+								app.destroy();
+								res(result);
```

Comment:
Whats this stuff?

Reply:
oh of course, my bad.

The only problem is it was implemented this way because i was having an issue receiving message out of order, which isn't technically supposed to be possible but it still happened. So the original logic allowed either message to resolve the promise whereas this only allows the status function to finish it, which if it is processed _before_ the final data message means that the return value of predict would be the second to last value.

Maybe we could use the `data.completed` property and add the dual handling back?

Reply:
Ok I added the dual handling back